src/main/java/org/hpccsystems/jdbcdriver/HPCCJDBCUtils.java

```diff
@@ -797,8 +797,15 @@ else if (java.math.BigDecimal.class == clazz)
             return new BigDecimal(value.replaceAll(",", ""));
         else if (java.sql.Date.class == clazz) //ASSUMES yyyy-MM-dd!!!
             return java.sql.Date.valueOf(value);
-        else if (java.sql.Time.class == clazz)//ASSUMES HH:mm 12 hour format!!!
-            return new java.sql.Time(((java.util.Date)HoursMinutesSeconds12HourFormat.parse(value)).getTime());
+        else if (java.sql.Time.class == clazz)//ASSUMES HH:mm:ss!!!
+        try
+        {
+            return new java.sql.Time(((java.util.Date)HoursColonMinutesColonSeconds24HourFormat.parse(value)).getTime());
```

Comment:
What happens here if the exception is thrown in the constructor or getTime() method?  What object type gets returned and is it guaranteed behavior.